Touch keyboard support: ability to configure how the key is pressed via hover up (tuch typing) or double-tap (standard typing)

josephsl opened the pull request:
Hi,

The following resolves the below issue:

* #7309: allow users to tell NVDA how one would interact with touch keyboard keys and send key presses, either via hover up (touch typing) or double tap (standard typing). A new dialog has been added to house this setting and future options.

Suggested what's new entry:

* Changes: On computers with touchscreens and with touch keyboard active, you need to double tap on a key instead of lifting your finger in order to press a key. This can be configured from a new Touch Interaction dialog found in preferences menu. (#7309)

Thanks.

josephsl committed:
Touch support: add a new map for touch interaction settings such as stnadard versus tyouch typing and what not. re #7309.

josephsl committed:
Global commands/hover up: press the touch key if touch typing option is enabled.

By default, touch typing will be disabled, but can be enabled by changing touch interaction settings.

josephsl committed:
Touchscreen/touch typing: add a new Touch Interaction dialog to configure touch typing mode between touch and standard.

Instead of just calling this 'touchscreen settings', touch interaction settings is a more appropriate title as it may include other interaction methods and settings in the future.

josephsl committed:
User guide: document the new touch interaction dialog and added a section on using the touch keyboard.

josephsl committed:
Touch interaction: only add this dialog if and only if touchscreen support is in use. re #7309

josephsl committed:
Fix translator comment mispelling.

josephsl committed:
Touch Interaction: make sure to call the correct super function for the correct dialog class.

michaelDCurran commented on user_docs/en/userGuide.t2t:
```diff
@@ -220,6 +220,14 @@ If the command does not have a mode listed, it will work in any mode.
 
++++ Touch keyboard +++
+The touch keyboard is used to enter text and commands from a touchscreen.
+When focused on an edit field, you can bring up the touch keyboard by double-tapping the touch keyboard icon on the bottom of the screen.
```
If the device is in Tablet mode, which may happen when no physical keyboard is connected to the machine (E.g. a Microsoft Surface Pro with its keyboard detached) the touch keyboard appears automatically when edit fields are focused.

josephsl committed:
Merge branch 'master' into i7309touchTyping

josephsl committed:
User guide: includethe fact that tablets display touch keyboard when keyboards are undocked. re #7309.

A reminder from Mick Curran (NV Access): for tablets such as Microsoft surface Pro and others, when the computer is in tablet mode or keyboard is undocked, touch keyboard becomes visible and stays on top at all times.

michaelDCurran committed:
Merge branch 'master' into i7309touchTyping

michaelDCurran commented on source/gui/__init__.py:
```diff
@@ -344,6 +348,10 @@ def __init__(self, frame):
 		# Translators: The label for the menu item to open Mouse Settings dialog.
 		item = menu_preferences.Append(wx.ID_ANY, _("&Mouse settings..."),_("Change reporting of mouse shape and object under mouse"))
```
There need to be two separate translator strings here for scons tests to pass.
Something like:
```
			item = menu_preferences.Append(wx.ID_ANY, 
				# Translators: The label for the menu item to open the Touch Interaction dialog.
				_("&Touch interaction..."),
				# Translators: The description for the menu item to open the Touch Interaction dialog.
				_("Change how NVDA interacts with the touchscreen")
			)
```

josephsl committed:
Touch interaction dialog: add translator comment for the tooltip.

A word from Mick Curran (NV Access) to add translator comments in order for scons checkPot to pass.

josephsl committed:
Touch handler and GUI: check if touch is supported, as touch handler might not be ready when initializing GUI subsystem. re #7309.

Touch handler is initialized AFTER GUI, so handler might not be ready at all. To avoid this, a new touchHandler.touchSupported function will be used to check if touch interaction is supported. For now, it is used by GUI, but add-ons and other modules are more than welcome to use it."

josephsl committed:
Touch handler: docstring

michaelDCurran reviewed (changes requested):
As the call to GetSystemMetrics is duplicated now (which is okay), can I ask that you define SM_MAXIMUMTOUCHES=95 at the top of touchHandler and use that constant in both places rather than 95 directly.

josephsl committed:
Touch handler: use max touches constant (95) as a proper constant. re #7309.

Noted by Mick Curran (NV Access): because the literal '95' is used multiple times, give it a proper label and use it as a constant.
